src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/CustomAttributes/RuntimeCustomAttributeData.cs

```diff
@@ -166,13 +167,13 @@ protected static CustomAttributeTypedArgument WrapInCustomAttributeTypedArgument
             }
 
             // Handle the array case
-            if (value is IEnumerable enumerableValue && !(value is string))
+            if (value is Array arr)
```

Comment:
This single change has biggest impact (-7.6 KB in hello world with reflection!). It causes enumerator of every collection considered as potentially boxed. By removing this, many of the enumerators can be trimmed. It also reduces the size gap between Dictionary implementations.

However, this can introduce a trap for user, that using weakly-type `IEnumerbale` can have increased size penalty.

Reply:
Extracted to #117345 for higher confidence.